molgenis-data-rest/src/main/java/org/molgenis/data/rest/v2/RestControllerV2.java

```diff
@@ -234,6 +234,7 @@ public void deleteEntityCollection(@PathVariable("entityTypeId") String entityTy
 	 * Retrieve an entity collection, optionally specify which attributes to include in the response.
 	 */
 	@GetMapping("/{entityTypeId}")
+	@Transactional(readOnly = true)
```

Comment:
How about marking the DataServiceImpl.findAll with ```@Transactional(readOnly=true, propagation = MANDATORY)``` to enforce this behavior everywhere?